test/unit/firebase-namespace.spec.ts

```diff
@@ -550,4 +551,35 @@ describe('FirebaseNamespace', () => {
       expect(firebaseNamespace.firestore.setLogFunction).to.be.deep.equal(setLogFunction);
     });
   });
+
+  describe('#instanceId()', () => {
+    it('should throw when called before initializing an app', () => {
+      expect(() => {
+        firebaseNamespace.instanceId();
+      }).to.throw(DEFAULT_APP_NOT_FOUND);
+    });
+
+    it('should throw when default app is not initialized', () => {
+      firebaseNamespace.initializeApp(mocks.appOptions, 'testApp');
+      expect(() => {
+        firebaseNamespace.instanceId();
+      }).to.throw(DEFAULT_APP_NOT_FOUND);
+    });
+
+    it('should return a valid namespace when the default app is initialized', () => {
+      let app: FirebaseApp = firebaseNamespace.initializeApp(mocks.appOptions);
+      let iid: InstanceId = firebaseNamespace.instanceId();
+      expect(iid).to.not.be.null;
```

Comment:
Add check:
expect(iid.app).to.be.deep.equal(app);

Reply:
Done